batch/batch/cloud/gcp/resources.py

```diff
@@ -202,6 +202,46 @@ def to_dict(self) -> dict:
         return {'type': self.TYPE, 'name': self.name, 'format_version': self.FORMAT_VERSION}
 
 
+class GCPAcceleratorResourceV2(VMResourceMixin, GCPResource):
+    FORMAT_VERSION = 1
```

Comment:
I can see the reasoning for making a new class here, but these resources are already versioned using this `FORMAT_VERSION` constant. Instead of creating a copy class, you should be able to change the existing accelerator class by adding the `number` field and incrementing `FORMAT_VERSION`. 

The reason the versioning is critical is because `to_dict` could be called with a dictionary that was generated before this code is merged in to main. So as long as `to_dict` reads the format version and defaults to 1 GPU when `FORMAT_VERSION` is 1 this should be backwards compatible